src/containers/RestaurantsList.js

```diff
@@ -0,0 +1,61 @@
+import React,{ Component } from 'react';
+import '../Styles.css'
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {fetchRestaurants} from '../actions/Actions';
+import Button from '@material-ui/core/Button';
+import GridView from '../components/GridView.js';
+
+class RestaurantsList extends Component{
+
+    componentDidMount() {
+       this.props.getRestaurants();
+    }
+
+    render(){
+        const { restaurants, isFetching, lastUpdate, error } = this.props;
+        return(
+            <div className='Restaurants-list'>
+                <p>
+                    {!isFetching &&
+                    (<Button variant='contained'
+                             size='small'
+                             color='primary'
+                             onClick={() => this.props.getRestaurants()}>
+                        Refresh
+                    </Button>)}
+                    {error && (<span>{error.message}</span>)}
+                    {lastUpdate && !error && (<span>Last updated at {new Date(lastUpdate).toLocaleTimeString()}</span>)}
+                </p>
+                <GridView restaurants={restaurants}/>
```

Comment:
you could have conditional rendering here, since you told me you got issues with the default value.
`restaurants && (<GridView restaurants={restaurants} />)`

Reply:
I know, but putting a guard here might obscure the right flow and initialisation, anyways: I took the defaultState from the reducer and used it in the preloadedState of the store so there is no overhead now